src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs

```diff
@@ -1066,13 +1066,21 @@ public ValidateTransactionResult ValidateTransaction(Transaction transaction, bo
                 {
                     TransactionBuilder builder = new TransactionBuilder(this.Wallet.Network).AddCoins(coins);
 
-                    if (builder.Verify(transaction, transaction.GetFee(coins.ToArray()), out TransactionPolicyError[] errors))
+                    var verifyResult = builder.Verify(transaction, transaction.GetFee(coins.ToArray()), out TransactionPolicyError[] errors);
+                    if (verifyResult)
+                        return ValidateTransactionResult.Valid();
+
+                    // Ignore any fee related errors here as the BuildTransaction method in FederationWalletTransactionHandler would have already
```

Comment:
Its probably not sufficient to just check the transaction in `BuildTransaction` as it (and most of the wallet code) is not part of the low-level consensus amongst nodes. Please ensure that there are some rules in place that enforce what federation transactions should look like in regards to fees.